### PR TITLE
Add Twitter link to navbar

### LINKED ISSUE #80: Add link to Twitter
Add link to our Twitter profile. Probably in the nav bar?

### PR DESCRIPTION
**Pull request by Steellow**
Fixes #80

### CODE REVIEW

File: src/js/components/Base.js [hunk 1]
```diff
@@ -21,14 +28,15 @@ const Header = ({ currentUrl }) => {
...
-            ({ href, title }) => html`
+            ({ href, title, external }) => html`
               <li class="inline">
                 <${Link}
                   class=${tw(
                     linkClasses,
                     currentUrl && currentUrl.startsWith(href) && 'font-bold'
                   )}
                   href=${href}
+                  target=${external ? '_blank' : '_self'}
```
[the remark this answers is not in the document]
Comment (Steellow) [reply src/js/components/Base.js:39]: I've always learned that external links should be opened in new tab so didn't even think about it. Now that I thought about it, I still think that external links should be opened in a new tab 😁 For a regular user (like our mom) it can be confusing if they're browsing a website, click a link and suddenly they are on another website. Example:

1. Open kk.fi
2. Click 'Twitter'
3. Browse our Twitter account
4. See interesting retweet, go see someones profile etc, do some browsing inside Twitter
5. You're done with Twitter and close the tab, and suddenly you can't see the kk.fi tab anywhere??

Of course you and I know we can spam the back-button or go to kk.fi again, but that might be too big of a step for a not-so-tech-savvy person. Especially if they arrived on your website on accident and can't remember the url. Doesn't really matter in this case, but is a big deal if you want to keep the user engaging with your site as long as possible (more ad revenue oslt.)

That was basically a long way of saying that I think you should never close your own site on behalf of the user. However as a power-user I agree that it's irritating when a link opens in new tab when you don't want it to. And our website visitors will anyway be nerds so I agree that Twitter should not be opened in a new tab 😂

I'll make the change

**Comment by Steellow** [comment]
> 
> 
> The changes to `package-lock.json` are off-topic, so maybe revert them?

Idk shouldn't package-lock be just commited every time and not be messed with? I probably ran `npm i` which updated some child dependencies

File: src/js/components/Base.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mtsknn) [reply src/js/components/Base.js]: Agree about power users. 👍

I see your point about regular users. And I disagree. 😂 Alternative viewpoint: opening links in new tabs by default is how some users end up with dozens open tabs, hundreds even. I should ponder this more thoroughly by writing a blog post, but yea, we can anyway agree to disagree. 😀

File: src/js/components/Base.js [hunk 2]
```diff
@@ -7,8 +7,13 @@ import Link from './Link'
...
 const Header = ({ currentUrl }) => {
-  const linkClasses = 'no-underline -mx-1 p-1 rounded hover:bg-gray-100'
-  const links = [{ href: '/info/', title: 'Info' }]
+  const linkClasses =
+    'no-underline -mx-1 p-1 rounded hover:bg-gray-100 whitespace-nowrap'
+  const links = [
+    { href: '/info/', title: 'Info' },
+    { href: 'https://twitter.com/koodikrapula', title: 'Twitter' },
+    { href: 'https://forms.gle/uyh3v8bbQUwnizEw5', title: 'Kysy kysymys' },
+  ]
```
Comment (mtsknn) [inline src/js/components/Base.js:16]: Idea: use the "external-link" icon from [Heroicons](https://heroicons.com/) after the external links' texts. Wdyt?

(Asking this question again because non-threaded questions are easy to miss.)

File: prettier.config.js [hunk 3]
```diff
@@ -3,4 +3,5 @@
 module.exports = {
   semi: false,
   singleQuote: true,
+  trailingComma: 'es5',
```
Comment (mtsknn) [inline prettier.config.js:6]: The default is `'es5'`: https://prettier.io/docs/en/options.html#trailing-commas

I would revert this unnecessary addition (but it doesn't do anything, so it's also OK to me to keep it).

**Review by mtsknn — CHANGES_REQUESTED** [review]
> Idk shouldn't package-lock be just commited every time and not be messed with?

`package-lock.json` should be committed, but only if you meant to introduce changes to the dependencies. Usually that would also mean updates to `package.json`.

I'd say that committing `package-lock.json` in other cases (when there shouldn't be updates to the dependencies) is better avoided to prevent accidental changes and bugs.

Modifying the lock file in non-relevant PRs also adds unnecessary fluff. (In this case: "2,658 changes: 1,322 additions & 1,336 deletions." I don't want to review these changes. 😅)

> I probably ran `npm i` which updated some child dependencies

Hmm, running `npm i` should install the packages specified in `package-lock.json` and _not_ modify it, so idk why that happened.

***

**TL;DR** Please revert the changes to `package-lock.json`. 👮‍♂️
